addons/binding/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/EvoCommunicator.java

```diff
@@ -0,0 +1,326 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.openhab.binding.paradoxalarm.internal.communication.messages.EpromRequestPayload;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.HeaderMessageType;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.IPPacketPayload;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.ParadoxIPPacket;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.RamRequestPayload;
+import org.openhab.binding.paradoxalarm.internal.exceptions.ParadoxBindingException;
+import org.openhab.binding.paradoxalarm.internal.model.ZoneStateFlags;
+import org.openhab.binding.paradoxalarm.internal.util.ParadoxUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EvoCommunicator} is responsible for handling communication to Evo192 alarm system via IP150 interface.
+ *
+ * @author Konstantin_Polihronov - Initial contribution
+ */
+public class EvoCommunicator extends GenericCommunicator implements IParadoxCommunicator {
+
+    protected final Logger logger = LoggerFactory.getLogger(EvoCommunicator.class);
+
+    private MemoryMap memoryMap;
+
+    public EvoCommunicator(String ipAddress, int tcpPort, String ip150Password, String pcPassword) throws Exception {
+        super(ipAddress, tcpPort, ip150Password, pcPassword);
+        initializeMemoryMap();
+    }
+
+    @Override
+    public List<String> readPartitionLabels() {
+        List<String> result = new ArrayList<>();
+
+        try {
+            for (int i = 1; i <= 8; i++) {
+                result.add(readPartitionLabel(i));
+            }
+        } catch (Exception e) {
+            logger.debug("Unable to retrieve partition labels. Exception: {}", e);
+        }
+        return result;
+    }
+
+    public String readPartitionLabel(int partitionNo) throws Exception {
+        logger.debug("Reading partition label: " + partitionNo);
+        if (partitionNo < 1 || partitionNo > 8) {
+            throw new Exception("Invalid partition number. Valid values are 1-8.");
+        }
+
+        int address = 0x3A6B + (partitionNo - 1) * 107;
+        byte labelLength = 16;
+
+        byte[] payloadResult = readEepromMemory(address, labelLength);
+
+        String result = createString(payloadResult);
+        logger.debug("Partition label: {}", result);
+        return result;
+    }
+
+    @Override
+    public List<String> readZoneLabels() {
+        List<String> result = new ArrayList<>();
+
+        try {
+            for (int i = 1; i <= 60; i++) {
+                result.add(readZoneLabel(i));
+            }
+        } catch (Exception e) {
+            logger.debug("Unable to retrieve zone labels. Exception: {}", e);
+        }
+        return result;
+    }
+
+    public String readZoneLabel(int zoneNumber) throws Exception {
+        logger.debug("Reading zone label: " + zoneNumber);
```

Comment:
```suggestion
        logger.debug("Reading zone label: {}", zoneNumber);
```
Note: Also check other log usages to not concatenate strings. Concatenating is always done even when the specific log level is not enabled, while using the variables as argument it's only evaluated when the specific log level is enabled.

Reply:
I understand this. Looked everywhere. Hopefully with next commit it will be OK.